ankaios_sdk/_components/control_interface.py

```diff
@@ -278,14 +278,13 @@ def _read_from_control_interface(self) -> None:
                         break
 
                 if not varint_buffer:
-                    if self._state != ControlInterfaceState.AGENT_DISCONNECTED:
-                        self.change_state(
-                            ControlInterfaceState.AGENT_DISCONNECTED)
+                    self.change_state(
+                        ControlInterfaceState.AGENT_DISCONNECTED)
                     self._logger.error(
                         "Nothing to read from the input fifo pipe. "
                         "Is the agent still there?"
```

Comment:
Maybe we can delete the second line here since we now output the log with the waiting.

Reply:
Done.